From: Niels Dekker <N.Dekker@lumc.nl>
Subject: Make itkOptImageToImageMetricsTest04 not to fail on single-CPU systems
Bug-Debian: https://bugs.debian.org/1105896
Origin: https://github.com/InsightSoftwareConsortium/ITK/commit/57b138cf150e097340b75e7f37eee061da80c627
Last-Update: 2025-05-16

--- a/Modules/Nonunit/Review/test/itkOptImageToImageMetricsTest2.h
+++ b/Modules/Nonunit/Review/test/itkOptImageToImageMetricsTest2.h
@@ -81,16 +81,6 @@
     // so we seed the random number generator
     // immediately before this call.
     metric->Initialize();
-
-    // Verify that Initialize has properly called
-    // MultiThreadingInitialize() and a series of CreateAnother();
-
-    using TransformPointer = typename MetricType::TransformPointer;
-    const TransformPointer * transformPtr = metric->GetThreaderTransform();
-    if ((transformPtr == static_cast<const TransformPointer *>(nullptr)) || (transformPtr[0].IsNull()))
-    {
-      exit(EXIT_FAILURE);
-    }
   }
   // Other registration functionality tested in
   // OptImageToImageTest.cxx... skip the rest
