Description: Add missing GL header
Author: Anton Gladky <gladk@debian.org>
Last-Update: 2024-01-10

--- vtk9-9.3.0+dfsg1.orig/GUISupport/Qt/QVTKOpenGLWindow.cxx
+++ vtk9-9.3.0+dfsg1/GUISupport/Qt/QVTKOpenGLWindow.cxx
@@ -22,6 +22,8 @@
 #include "vtkObjectFactory.h"
 #include "vtkOpenGLState.h"
 
+#include <GL/gl.h>
+
 VTK_ABI_NAMESPACE_BEGIN
 QVTKOpenGLWindow::QVTKOpenGLWindow(QOpenGLWindow::UpdateBehavior ub, QWindow* p)
   : QVTKOpenGLWindow(vtkSmartPointer<vtkGenericOpenGLRenderWindow>::New(), nullptr, ub, p)
